membership-attribute-service/test/services/SupporterProductDataIntegrationTest.scala

```diff
@@ -0,0 +1,113 @@
+package services
+
+import akka.actor.ActorSystem
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClientBuilder
+import com.gu.memsub.subsv2.SubscriptionPlan.AnyPlan
+import com.typesafe.scalalogging.LazyLogging
+import components.TouchpointComponents
+import models.{Attributes, ContentAccess}
+import org.specs2.concurrent.ExecutionEnv
+import org.specs2.matcher.MatchResult
+import org.specs2.mutable.Specification
+import play.api.libs.json.Json
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Future}
+import scala.io.Source
+
+class SupporterProductDataIntegrationTest(implicit ee: ExecutionEnv) extends Specification with LazyLogging {
+
+  val stage = "PROD" // Whichever stage is specified here, you will need config for it in /etc/gu/members-data-api.private.conf
+  lazy val dynamoClientBuilder: AmazonDynamoDBAsyncClientBuilder = AmazonDynamoDBAsyncClientBuilder.standard().withCredentials(com.gu.aws.CredentialsProvider).withRegion(Regions.EU_WEST_1)
+  lazy val mapper = new SupporterRatePlanToAttributesMapper(stage)
+  lazy val supporterProductDataTable = "SupporterProductData-PROD"
+  lazy val supporterProductDataService = new SupporterProductDataService(dynamoClientBuilder.build(), supporterProductDataTable, mapper)
+
+  lazy val dynamoAttributesTable = "SupporterAttributesFallback-PROD"
+  lazy val attrService: AttributeService = new ScanamoAttributeService(dynamoClientBuilder.build(), dynamoAttributesTable)
+
+
+  implicit private val actorSystem: ActorSystem = ActorSystem()
+  lazy val touchpoint = new TouchpointComponents("PROD")
+  lazy val attributesFromZuora = new AttributesFromZuora()
+
+  args(skipAll = true) // This test requires credentials so won't run on CI, change skipAll to false to run locally
+
+  "SupporterProductData" should {
+    "match Zuora" in {
+      val excluded = List()
+      val allIds: List[String] = Source.fromURL(getClass.getResource("/identityIds.txt")).mkString.split("\n").toList.filter(!excluded.contains(_))
```

Comment:
it's not risk of the century, but as a habit I'd rather keep the file outside the VCS root to avoid a chance of checking it in.
```suggestion
      val allIds: List[String] = Source.fromFile(new File("~/identityIds.txt")).mkString.split("\n").toList.filter(!excluded.contains(_))
```

Reply:
I've just removed it actually, I don't think it will be particularly useful going forward.